LoginActivity completed

Alberto-Cortes opened the pull request:
LoginAcitivity is now complete and allows users to log into Spotify and Parse, the app login proceeds only if both were done correctly.
Login currently sends to empty Camera Activity, this new activity is a work in progress.

Alberto-Cortes committed:
Initial project commit

Initial project commit containing IDE generated files to work on project.

Alberto-Cortes committed:
New comment on main activity

Comment used to test github for desktop

Alberto-Cortes committed:
Spotify API connection

App connected with Spotify's API correctly. Added their .aar file to the project. Most of the changes were done automatically by Android Studio, the files I did change were the gradle to include de dependency and the MainActivity class.

Alberto-Cortes committed:
LoginActivity Parse and Spotify elements

MainActivity renamed to LoginActivity.
App login will consist of 2 steps, step 1 is Spotify webView login, step 2 is Parse login or signup. Visual elements are completed, missing logic.

Alberto-Cortes committed:
LoginActivity funcionality

Users are now able to login both with Spotify and Parse.

Alberto-Cortes committed:
Login validation and CameraActivity creation

Login now checks if user has done both steps, both Spotify and Parse. If login is successful the user is sent to the CameraActivity, currently empty.
Also added comments on everything the LoginActivity does.

mbarbagallo commented on app/src/main/java/com/example/photostomusic/LoginActivity.java:
```diff
@@ -0,0 +1,136 @@
+    public final String TAG = this.getClass().getSimpleName();
+
+    // PSF variables required to interact with the Spotify Auth API
+    private static final String CLIENT_ID = "59a64c83ef024ea786df03a966505f91";
```
does this need to be hardcoded here? could you not store the client ID in the DB?

mbarbagallo commented on app/src/main/java/com/example/photostomusic/LoginActivity.java:
```diff
@@ -0,0 +1,136 @@
+                        new AuthenticationRequest.Builder(CLIENT_ID, AuthenticationResponse.Type.TOKEN, REDIRECT_URI);
+
+                // Empty string to generate token with all permissions
+                builder.setScopes(new String[]{""});
```
are you doing anything with this string? does it need to be a string object or can you just pass in `""`

mbarbagallo commented on app/src/main/java/com/example/photostomusic/LoginActivity.java:
```diff
@@ -0,0 +1,136 @@
+                    break;
+
+                // Most likely auth flow was cancelled
+                default:
```
maybe add a toast or a log on default?

mbarbagallo commented on app/src/main/java/com/example/photostomusic/ParseApplication.java:
```diff
@@ -0,0 +1,25 @@
+
+        Parse.initialize(new Parse.Configuration.Builder(this)
+                .applicationId("U0QJQJaso5fDH0ZzD0G8gF7y6nVcEfpfD67l81Ez")
+                .clientKey("XaHGTAgQZ78nZSaHmlZX5ppXwJ4WdMYirwRfMabm")
```
should the client key be public?

mbarbagallo reviewed (approved):
I added some comments. They are non-blockers on the PR, but some things to still keep in mind. I'm not sure having the string for things like the auth or client key be public in the classes like that is great.

Alberto-Cortes committed:
MainActivity navigation view

Renamed CameraActivity to MainActivity, Camera screen will now be inside a fragment accessible through the bottom navigation view.

mbarbagallo commented:
I'm closing this pull request because I think that i added everything by trying to resolve the merge conflicts.
